js/save.js

```diff
@@ -0,0 +1,51 @@
+/*  save.js
+ * 
+ *  This file lists down the functions used for save and load feature.
+ *  The triggers are defined in gui.js
+ */
+var saveObjectName = "labyrinthSaveObject";
+
+function gameState () {
+    this.badges = new badges();
+    this.inventory = new inventory();
+}
+
+function saveExists() {
+    if (localStorage.getItem(saveObjectName) == null) {
+        return 0;
+    } else {
+        return 1;
+    }
+}
+
+function saveGameState(saveObject) {
+    localStorage.setItem(saveObjectName, JSON.stringify(saveObject));
+    alert(JSON.stringify(saveObject));
+}
+
+function getSaveGame() {
+    var savedObjectJSONString = localStorage.getItem(saveObjectName);
+    var savedObject = JSON.parse(savedObjectJSONString);
+    return savedObject;
+}
+
+function saveGame(player) {
+    var state = new gameState();
+    state.badges = player.badges.items;
+    state.inventory = player.inventory.items;
```

Comment:
If you move the lines 34 and 35 into the player, this would be even better. https://en.wikipedia.org/wiki/Law_of_Demeter

Reply:
@niccokunzmann you mean like passing something like
```javascript
getItems(state) {
  state.badges = this.badges.items;
}
```